services/movie_session.py

```diff
@@ -0,0 +1,58 @@
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(movie_show_time: str,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    new_movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id
+    )
+    return new_movie_session
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    sessions = MovieSession.objects.all()
+
+    if session_date:
+        sessions = sessions.filter(show_time__year=session_date.split("-")[0],
+                                   show_time__month=session_date.split("-")[1],
+                                   show_time__day=session_date.split("-")[2])
+
+    return sessions
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    if MovieSession.objects.filter(id=movie_session_id).exists():
+        return MovieSession.objects.get(id=movie_session_id)
```

Comment:
Using `filter().exists()` followed by `get()` can lead to race conditions. It's better to use `get_object_or_404` or handle the `DoesNotExist` exception to ensure the application fails gracefully and provides user-friendly error messages.